emm/indexing/pandas_normalized_tfidf.py

```diff
@@ -74,6 +73,8 @@ def fit(self, X: pd.Series | pd.DataFrame) -> TfidfVectorizer:
         with Timer("CustomizedTfidfVectorizer.fit") as timer:
             timer.label("super fit")
             super().fit(X)
+            # scikit-learn's TfidfVectorizer does not preserve dtype for large X, so we force it here
+            self.idf_ = self.idf_.astype(self.dtype)
```

Comment:
So we have a fix in https://github.com/scikit-learn/scikit-learn/pull/30022.
Note that this only needed for NumPy < 2 then.

Reply:
That's great, thank you for the quick investigation!

Reply:
Thanks @glemaitre, good to know!